spec/system/admin/reports/sales_tax/sales_tax_totals_by_order_spec.rb

```diff
@@ -478,10 +478,16 @@ def extract_xlsx_rows(file, range)
           expect(downloaded_filenames.length).to eq(1) # downloads folder should contain 1 file
           expect(downloaded_filename).to match(/.*\.pdf/)
           downloaded_file_pdf = PDF::Reader.new(downloaded_filename)
-          # comparing pdf md5
-          md5_fixture = `cat report_file_pdf | grep -a -v "/CreationDate" | md5sum`.strip
-          md5_downloaded = `cat downloaded_file_pdf | grep -a -v "/CreationDate" | md5sum`.strip
-          expect(md5_fixture).to eql md5_downloaded
+
+          downloaded_file_txt = capture_io do
+            puts downloaded_file_pdf.pages.map(&:text)
+          end
```

Comment:
I think that this could be a lot simpler. Instead of taking the text as it is, you are writing it to the console with with `puts` and then "capture" that output again to recover it. The `puts` command does not return the string. It writes it to the output and doesn't return anything to the Ruby code. Maybe that's where you got confused before.